### PR TITLE
Enhancement | basespace pagination | #1wv013p

### PR DESCRIPTION
**Pull request by yairfischer**

### CODE REVIEW

**Comment by nivm** [comment]
Task linked: [CU-1wv013p BaseSpace V2 - pagination](https://app.clickup.com/t/1wv013p)

File: fs_basespace/basespace_context.py [hunk 1]
```diff
@@ -51,11 +54,12 @@ def __init__(self, raw_obj):
         self.raw_obj = raw_obj
 
     @abstractmethod
-    def list_raw(self, api: BasespaceApiFactory):
+    def list_raw(self, api: BasespaceApiFactory, page: Tuple[int, int]):
         raise NotImplementedError("Should return list of entity contexts")
 
-    def list(self, api: BasespaceApiFactory):
-        return [self.ENTITY_CONTEXT(entity) for entity in self.list_raw(api)]
+    # def list(self, api: BasespaceApiFactory):
```
Comment (amit-ezra) [inline fs_basespace/basespace_context.py:60]: please revmove comments

File: fs_basespace/basespace_context.py [hunk 2]
```diff
@@ -25,7 +27,8 @@ class EntityContext(metaclass=EntityContextMeta):
     def __init__(self, raw_obj):
         self.raw_obj = raw_obj
 
-    def list(self, api: BasespaceApiFactory):
+    # def list(self, api: BasespaceApiFactory, page: Tuple[int, int]):
+    def list(self, api: BasespaceApiFactory, page: Tuple[int, int]):
```
Comment (amit-ezra) [inline fs_basespace/basespace_context.py:31]: 1. I like that we use type hinitng. makes everything more readable. 
2. more readable for these type of type hining is to use aliases 
[see](https://docs.python.org/3/library/typing.html#type-aliases)
basically define 
somewhere that 
`Page = Tuple[int, int]`
than everywhere you can do
`page : Page`

File: fs_basespace/basespace_context.py [hunk 3]
```diff
@@ -263,15 +277,23 @@ def get_last_direct_context(key):
...
         path_steps = rest_path.split("/")
-        latest_context = latest_context_cls.get_entity_direct(api, path_steps[0])
+        latest_context = latest_context_cls.get_entity_direct(api, path_steps[0], page)
         rest_steps = path_steps[1:]
     for path_step in rest_steps:
         latest_context = latest_context.get(api, path_step)
     return latest_context
+
+
+def translate_range_to_offset_and_limit(page: Tuple[int, int]):
```
Comment (amit-ezra) [inline fs_basespace/basespace_context.py:294]:
```suggestion
def translate_page_to_offset_and_limit(page: Tuple[int, int]):
```

Comment (amit-ezra) [reply fs_basespace/basespace_context.py:294]: as your input is page and not a range

**Review by amit-ezra — APPROVED** [review]

**Pull request merged**